test/bdd/exec_test.go

```diff
@@ -0,0 +1,334 @@
+		It("should create a CustomResource ChaosExperiment Pod-delete", func() {
+
+			By("building a chaosExperiment")
+			ChaosExperiment := &v1alpha1.ChaosExperiment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pod-delete",
+					Namespace: "litmus",
+					Labels: map[string]string{
+						"litmuschaos.io/name": "kubernetes",
+					},
+				},
+				Spec: v1alpha1.ChaosExperimentSpec{
+					Definition: v1alpha1.ExperimentDef{
+
+						Args:    []string{"-c", "ansible-playbook ./experiments/generic/pod_delete/pod_delete_ansible_logic.yml -i /etc/ansible/hosts -vv; exit 0"},
+						Command: []string{"/bin/bash"},
+						Image:   "litmuschaos/ansible-runner:ci",
+						ENVList: []v1alpha1.ENVPair{
+							{
+								Name:  "ANSIBLE_STDOUT_CALLBACK",
+								Value: "default",
+							},
+							{
+								Name:  "TOTAL_CHAOS_DURATION",
+								Value: "15",
+							},
+							{
+								Name:  "CHAOS_INTERVAL",
+								Value: "5",
+							},
+							{
+								Name:  "LIB",
+								Value: "",
+							},
+						},
+						Labels: map[string]string{
+							"name": "pod-delete",
+						},
+					},
+				},
+			}
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building pod-delete chaosExperiment",
+			)
+
+			By("creating above chaosExperiment")
+			_, err = clientSet.ChaosExperiments("litmus").Create(ChaosExperiment)
+			Expect(err).To(
+				BeNil(),
+				"while creating chaosExerpiment Pod-delete in namespace litmus",
+			)
+		})
+	})
+	When("Creating ChaosEngine to trigger chaos-executor", func() {
+		It("should create a runnerPod and Service ", func() {
+
+			By("building a ChaosEngine")
+			chaosEngine := &v1alpha1.ChaosEngine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "engine-nginx",
+					Namespace: "litmus",
+				},
+				Spec: v1alpha1.ChaosEngineSpec{
+					Components: v1alpha1.ComponentParams{
+						Runner: v1alpha1.RunnerInfo{
+							Type:  "go",
+							Image: "rahulchheda1997/chaos-executor:ci",
+						},
+					},
+					Appinfo: v1alpha1.ApplicationParams{
+						Appns:    "litmus",
+						Applabel: "app=nginx",
+						AppKind:  "deployment",
+					},
+					ChaosServiceAccount: "litmus",
+					Monitoring:          true,
+					Experiments: []v1alpha1.ExperimentList{
+						{
+							Name: "pod-delete",
+						},
+					},
+				},
+			}
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building ChaosEngine testEngine in namespace litmus",
+			)
+
+			By("Creating above ChaosEngine")
+			_, err = clientSet.ChaosEngines("litmus").Create(chaosEngine)
+			Expect(err).To(
+				BeNil(),
+				"while building ChaosEngine testEngine in namespace litmus",
+			)
+
+			time.Sleep(30 * time.Second)
+
+			//Fetching engine-nginx-runner pod
+			runner, err := client.CoreV1().Pods("litmus").Get("engine-nginx-runner", metav1.GetOptions{})
+			//Fetching engine-nginx-exporter pod
+			exporter, err := client.CoreV1().Pods("litmus").Get("engine-nginx-monitor", metav1.GetOptions{})
```

Comment:
Check for err

Reply:
No need. it will automatically be shown once the nil check is turned out to be false.